src/test/java/teammates/test/cases/common/TimeHelperTest.java

```diff
@@ -92,9 +92,9 @@ public void testEndOfYearDates() {
         cal.set(2015, 11, 30, 12, 0, 0);
         Date date = cal.getTime();
         assertEquals("30/12/2015", TimeHelper.formatDate(date));
-        assertEquals("Wed, 30 Dec 2015, 12:00 PM", TimeHelper.formatTime12H(date));
+        assertEquals("Wed, 30 Dec 2015, 12:00 NOON", TimeHelper.formatTime12H(date));
         assertEquals("Wed, 30 Dec 2015, 12:00 PM UTC", TimeHelper.formatDateTimeForComments(date));
-        assertEquals("30 Dec 12:00 PM", TimeHelper.formatDateTimeForInstructorHomePage(date));
+        assertEquals("30 Dec 12:00 NOON", TimeHelper.formatDateTimeForInstructorHomePage(date));
```

Comment:
Sorry, but I think you can stick to the initial behaviour that you had then. For 12 noon, show it as "12 NOON", without the minutes. It is fine to show the minutes for 12:01pm, as mentioned in the other comment. For midnight, there is no need for special behaviour and can be displayed as 0 am. Sorry for the changing requirements.